dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/rest/TelemetryContainerFilter.java

```diff
@@ -145,34 +145,22 @@ public void filter(final ContainerRequestContext request) {
                 Class<?> resourceClass = resourceInfo.getResourceClass();
                 Method resourceMethod = resourceInfo.getResourceMethod();
 
-                String route = ROUTE_CACHE.getRoute(resourceClass, resourceMethod);
+                String route = getRoute(request, resourceClass, resourceMethod);
 
-                if (route == null) {
-
-                    String contextRoot = request.getUriInfo().getBaseUri().getPath();
-                    UriBuilder template = UriBuilder.fromPath(contextRoot);
-
-                    if (resourceClass.isAnnotationPresent(Path.class)) {
-                        template.path(resourceClass);
-                    }
-
-                    if (resourceMethod.isAnnotationPresent(Path.class)) {
-                        template.path(resourceMethod);
-                    }
-
-                    route = template.toTemplate();
-                    ROUTE_CACHE.putRoute(resourceClass, resourceMethod, route);
+                if (route != null) {
+                    currentSpan.setAttribute(SemanticAttributes.HTTP_ROUTE, route);
+                    currentSpan.updateName(request.getMethod() + " " + route);
                 }
 
                 currentSpan.setAttribute(SemanticAttributes.HTTP_ROUTE, route);
-                currentSpan.updateName(request.getMethod()  + " " + route);
+                currentSpan.updateName(request.getMethod() + " " + route);
```

Comment:
You're setting the attribute and the name both inside and outside of your if block